crates/sui-core/src/storage.rs

```diff
@@ -338,3 +340,151 @@ impl WriteStore for RocksDbStore {
         Ok(())
     }
 }
+
+pub struct RestReadStore {
+    state: Arc<AuthorityState>,
+    rocks: RocksDbStore,
+}
+
+impl RestReadStore {
+    pub fn new(state: Arc<AuthorityState>, rocks: RocksDbStore) -> Self {
+        Self { state, rocks }
+    }
+}
+
+impl ObjectStore for RestReadStore {
+    fn get_object(
+        &self,
+        object_id: &sui_types::base_types::ObjectID,
+    ) -> sui_types::storage::error::Result<Option<Object>> {
+        self.rocks.get_object(object_id)
+    }
+
+    fn get_object_by_key(
+        &self,
+        object_id: &sui_types::base_types::ObjectID,
+        version: sui_types::base_types::VersionNumber,
```

Comment:
What's the prevailing sentiment on `VersionNumber` vs `SequenceNumber`? I don't really mind which we use, but it would be good to be consistent.